src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CMDLineProgramModule.java

```diff
@@ -0,0 +1,124 @@
+ */
+public abstract class CMDLineProgramModule {
+
+    public CMDLineProgramModule(){  }
+
+    /**
+     * Builds the initial argument list for the command line (e.g. "bwa mem").
+     * @return a list of strings having the names of requested tools.
+     * @throws IOException
+     */
+    abstract ArrayList<String> initializeCommands() throws IOException;
+
+    /**
+     * Worker function to call the cmd line program.
+     * @param runtimeArguments          arguments to be provided to the program
+     * @param directoryToWorkIn         a directory for the program to work in (mainly for output)
+     * @param stdoutDestination         redirects the stdout from the program to a file
+     * @param stderrMessageDestination  redirects the stderr message to a string
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws RuntimeException
+     */
+    public void run(final ArrayList<String> runtimeArguments,
+                    final File directoryToWorkIn,
+                    final File stdoutDestination,
+                    String stderrMessageDestination) throws IOException, InterruptedException, RuntimeException{
+
+        final ArrayList<String> commands = initializeCommands();
+        for(final String arg : runtimeArguments){ commands.add(arg); }
+
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        setupWorkingDir(builder, directoryToWorkIn, stdoutDestination);
+        setupWorkingEnvironment(builder);
+
+        Process runProcess = builder.start();
+        int exitStatus = runProcess.waitFor();
+        stderrMessageDestination = redirectSTDERR(runProcess);
+
+        if(0!=exitStatus){ onError(commands, stderrMessageDestination, exitStatus); }
+    }
+
+    protected void setupWorkingDir(final ProcessBuilder builder,
+                                   final File directoryToWorkIn,
+                                   final File stdoutDestination){
+        builder.directory(directoryToWorkIn);
+        if(null!=stdoutDestination) { builder.redirectOutput(stdoutDestination); }
+    }
+
+    /**
+     * Sets up working environment for the program. Inheriting classes should provide implementation.
+     * @param builder process builder for the program
+     * @param args    arguments used for setting up the environment
+     */
+    abstract void setupWorkingEnvironment(final ProcessBuilder builder, final String... args);
+
+    /**
+     * Redirects the stderr message from the represented program to a string.
+     * @param process       the program's process
+     * @return              String converted from the stderr message of the program
+     * @throws IOException
+     */
+    protected static String redirectSTDERR(final Process process) throws IOException{
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()));
```

Comment:
reader is not closed. use a try-with-resources like this
`try(final BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()))){
...
}`

the reader will be closed